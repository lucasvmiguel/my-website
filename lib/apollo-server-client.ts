import { ApolloClient, InMemoryCache } from '@apollo/client'

export const client = new ApolloClient({
  uri: process.env.HASURA_APOLLO_CLIENT_URI,
  cache: new InMemoryCache(),
  headers: {
    'X-Hasura-Admin-Secret': process.env.HASURA_APOLLO_CLIENT_TOKEN as string
  }
})
