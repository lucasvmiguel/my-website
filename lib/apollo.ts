import { ApolloClient, InMemoryCache } from '@apollo/client'

export const client = new ApolloClient({
  cache: new InMemoryCache(),
  uri: `${process.env.NEXT_PUBLIC_BASE_URL}/api/graphql`
})
