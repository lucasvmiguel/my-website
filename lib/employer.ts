export interface Employer {
  name: string;
  description: string;
  image: string;
  jobTitle: string;
  tags: string[];
}

export const employers: Array<Employer> = [
  {
    name: 'Workday',
    description: 'Finance, HR, and planning',
    image: 'workday-logo.png',
    jobTitle: 'Software Engineer',
    tags: ['Kubernetes', 'Go', 'Helm', 'Istio', 'Terraform', 'Ruby', 'Ruby on Rails', 'Sidekiq', 'React', 'AWS', 'Docker']
  },
  {
    name: 'Noa',
    description: 'Journalism over audio',
    image: 'noa-logo.png',
    jobTitle: 'Software Engineer',
    tags: ['React', 'Redux', 'NodeJS', 'React Router', 'Google Analytics', 'Elasticsearch', 'MySQL', 'Express', 'Sequelize', 'Redux Form', 'AWS']
  },
  {
    name: 'B2W',
    description: 'E-commerce',
    image: 'b2w-logo.png',
    jobTitle: 'Software Engineer',
    tags: ['Go', 'Elasticsearch', 'Clojure', 'React', 'Redux', 'AngularJS', 'Microservices', 'Mongo', 'AWS', 'Docker']
  }
]
