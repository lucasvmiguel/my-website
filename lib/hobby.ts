import { IconType } from 'react-icons'
import { BiRun, BiCodeBlock } from 'react-icons/bi'
import { FaLuggageCart } from 'react-icons/fa'

export interface Hobby {
  title: string;
  description: string;
  icon: IconType;
}

export const hobbies: Array<Hobby> = [
  {
    title: 'Running',
    description: 'While I\'m listening music or a podcast is such a wonderful experience.',
    icon: BiRun
  },
  {
    title: 'Travel',
    description: 'Exploring the world and all its culture is something that always up to. My next destines are Egypt, Italy and Spain.',
    icon: FaLuggageCart
  },
  {
    title: 'Coding',
    description: 'Well, as you can see, I didn\'t need to have this website at all. However, I love coding in my free time.',
    icon: BiCodeBlock
  }
]
