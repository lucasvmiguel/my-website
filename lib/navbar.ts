export interface NavItem {
  label: string;
  subLabel?: string;
  children?: Array<NavItem>;
  href?: string;
}

export const navItems: Array<NavItem> = [
  {
    label: 'About me',
    href: '#'
  },
  {
    label: 'Articles',
    children: [
      {
        label: 'Article title 1',
        subLabel: 'Article description 1',
        href: '#'
      },
      {
        label: 'Article title 2',
        subLabel: 'Article description 2',
        href: '#'
      }
    ]
  }
]
