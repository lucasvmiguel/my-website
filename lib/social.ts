import { IconType } from 'react-icons'
import { FaLinkedin, FaTwitter } from 'react-icons/fa'
import { AiFillGithub } from 'react-icons/ai'

export interface SocialItem {
  label: string;
  icon: IconType;
  href: string;
}

export const socialItems: Array<SocialItem> = [
  {
    label: 'Github',
    icon: AiFillGithub,
    href: 'https://github.com/lucasvmiguel/'
  },
  {
    label: 'Linkedin',
    icon: FaLinkedin,
    href: 'https://www.linkedin.com/in/lucasvieiramiguel/'
  },
  {
    label: 'Twitter',
    icon: FaTwitter,
    href: 'https://twitter.com/LucasVieiraDev'
  }
]
