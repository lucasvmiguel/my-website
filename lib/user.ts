import { createUserMutation, findUserByExternalIdQuery } from '../graphql/user'

export const createOrFindUserIdByExternalId = async (externalId: string) => {
  const user = await findUserByExternalId(externalId)

  if (user) {
    return user.id
  }

  const { id } = await createUser(externalId)
  return id
}

const findUserByExternalId = async (externalId: string) => {
  const { data } = await findUserByExternalIdQuery(externalId)
  if (data.users.length > 0) {
    return data.users[0]
  }
  return null
}

const createUser = async (externalId: string) => {
  const { data: { insert_users_one: { id } } } = await createUserMutation('user', externalId)

  return { id }
}
