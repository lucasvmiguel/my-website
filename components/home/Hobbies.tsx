import { ReactElement } from 'react'
import { Box, SimpleGrid, Icon, Text, Stack, Flex, Heading, Container } from '@chakra-ui/react'
import { Hobby } from '../../lib/hobby'

interface FeatureProps {
  title: string;
  text: string;
  icon: ReactElement;
}

interface HobbiesProps {
  items: Hobby[];
}

const Feature = ({ title, text, icon }: FeatureProps) => {
  return (
    <Stack>
      <Flex
        w={16}
        h={16}
        align={'center'}
        justify={'center'}
        color={'white'}
        rounded={'full'}
        bg={'gray.100'}
        mb={1}>
        {icon}
      </Flex>
      <Text fontWeight={600}>{title}</Text>
      <Text color={'gray.600'}>{text}</Text>
    </Stack>
  )
}

const Hobbies = ({ items }: HobbiesProps) => {
  return (
    <Box p={4}>
      <Container maxW={'7xl'} py={16} as={Stack} spacing={12}>
        <Stack spacing={0} align={'center'}>
          <Heading>Hobbies</Heading>
          <Text>Things that I like to do</Text>
        </Stack>
        <SimpleGrid columns={{ base: 1, md: 3 }} spacing={10}>
          {items.map((item, i) => (
            <Feature
              key={i}
              icon={<Icon as={item.icon} w={10} h={10} color="green" />}
              title={item.title}
              text={
                item.description
              }
            />
          ))}
        </SimpleGrid>
      </Container>
    </Box>
  )
}

export default Hobbies
