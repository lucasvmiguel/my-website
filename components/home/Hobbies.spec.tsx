import renderer from 'react-test-renderer'

import Hobbies from './Hobbies'

import { hobbies } from '../../lib/hobby'

it('renders Hobbies component correctly - snapshot', () => {
  const tree = renderer
    .create(<Hobbies items={hobbies} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
