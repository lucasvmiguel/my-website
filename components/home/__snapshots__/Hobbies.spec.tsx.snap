// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Hobbies component correctly - snapshot 1`] = `
<div
  className="css-2rlxtj"
>
  <div
    className="chakra-stack chakra-container css-lxy0v2"
  >
    <div
      className="chakra-stack css-jp20jv"
    >
      <h2
        className="chakra-heading css-0"
      >
        Hobbies
      </h2>
      <p
        className="chakra-text css-0"
      >
        Things that I like to do
      </p>
    </div>
    <div
      className="css-79z96f"
    >
      <div
        className="chakra-stack css-n21gh5"
      >
        <div
          className="css-x64a6d"
        >
          <svg
            className="chakra-icon css-n3ais1"
            fill="currentColor"
            focusable={false}
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="17"
              cy="4"
              r="2"
            />
            <path
              d="M15.777,10.969c0.376,0.563,1.008,0.89,1.666,0.89c0.16,0,0.322-0.02,0.482-0.06l3.316-0.829L20.758,9.03l-3.316,0.829 l-1.379-2.067c-0.291-0.439-0.756-0.751-1.272-0.854l-3.846-0.77c-0.888-0.181-1.778,0.265-2.181,1.067l-1.658,3.316l1.789,0.895 l1.658-3.317l1.967,0.394L7.434,17H3v2h4.434c0.698,0,1.355-0.372,1.715-0.971l1.918-3.196l5.169,1.034l1.816,5.449l1.896-0.633 l-1.815-5.448c-0.226-0.679-0.802-1.188-1.506-1.33l-3.039-0.607l1.772-2.954L15.777,10.969z"
            />
          </svg>
        </div>
        <p
          className="chakra-text css-35ezg3"
        >
          Running
        </p>
        <p
          className="chakra-text css-djevj3"
        >
          While I'm listening music or a podcast is such a wonderful experience.
        </p>
      </div>
      <div
        className="chakra-stack css-n21gh5"
      >
        <div
          className="css-x64a6d"
        >
          <svg
            className="chakra-icon css-n3ais1"
            fill="currentColor"
            focusable={false}
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 640 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224 320h32V96h-32c-17.67 0-32 14.33-32 32v160c0 17.67 14.33 32 32 32zm352-32V128c0-17.67-14.33-32-32-32h-32v224h32c17.67 0 32-14.33 32-32zm48 96H128V16c0-8.84-7.16-16-16-16H16C7.16 0 0 7.16 0 16v32c0 8.84 7.16 16 16 16h48v368c0 8.84 7.16 16 16 16h82.94c-1.79 5.03-2.94 10.36-2.94 16 0 26.51 21.49 48 48 48s48-21.49 48-48c0-5.64-1.15-10.97-2.94-16h197.88c-1.79 5.03-2.94 10.36-2.94 16 0 26.51 21.49 48 48 48s48-21.49 48-48c0-5.64-1.15-10.97-2.94-16H624c8.84 0 16-7.16 16-16v-32c0-8.84-7.16-16-16-16zM480 96V48c0-26.51-21.49-48-48-48h-96c-26.51 0-48 21.49-48 48v272h192V96zm-48 0h-96V48h96v48z"
            />
          </svg>
        </div>
        <p
          className="chakra-text css-35ezg3"
        >
          Travel
        </p>
        <p
          className="chakra-text css-djevj3"
        >
          Exploring the world and all its culture is something that always up to. My next destines are Egypt, Italy and Spain.
        </p>
      </div>
      <div
        className="chakra-stack css-n21gh5"
      >
        <div
          className="css-x64a6d"
        >
          <svg
            className="chakra-icon css-n3ais1"
            fill="currentColor"
            focusable={false}
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20,3H4C2.897,3,2,3.897,2,5v14c0,1.103,0.897,2,2,2h16c1.103,0,2-0.897,2-2V5C22,3.897,21.103,3,20,3z M4,19V7h16 l0.002,12H4z"
            />
            <path
              d="M9.293 9.293L5.586 13 9.293 16.707 10.707 15.293 8.414 13 10.707 10.707zM14.707 9.293L13.293 10.707 15.586 13 13.293 15.293 14.707 16.707 18.414 13z"
            />
          </svg>
        </div>
        <p
          className="chakra-text css-35ezg3"
        >
          Coding
        </p>
        <p
          className="chakra-text css-djevj3"
        >
          Well, as you can see, I didn't need to have this website at all. However, I love coding in my free time.
        </p>
      </div>
    </div>
  </div>
</div>
`;
